src/lib/api/index.ts

```diff
@@ -313,3 +179,117 @@ export async function post(agent: BskyAgent, opts: PostOpts) {
 
   return res
 }
+
+async function resolveEmbed(
+  agent: BskyAgent,
+  opts: PostOpts,
+): Promise<
+  | AppBskyEmbedImages.Main
+  | AppBskyEmbedVideo.Main
+  | AppBskyEmbedExternal.Main
+  | AppBskyEmbedRecord.Main
+  | AppBskyEmbedRecordWithMedia.Main
+  | undefined
+> {
+  if (opts.extLink?.embed) {
+    return opts.extLink.embed
+  }
```

Comment:
This check (for no images) got lost here. So if you compose a post with an image(s), AND paste in a link for an embeddable entity like a Feed, the Feed is resolved and overwrites the images without any feedback to the user.

Existing functionality preserves the images, and the feed is left as a link facet.

Reply:
makes sense, thanks for explaining. the original behavior only did this for photos, but it makes sense to do for videos too. so i pushed that.